common/decklist.cpp

```diff
@@ -726,12 +726,24 @@ bool DeckList::deleteNode(AbstractDecklistNode *node, InnerDecklistNode *rootNod
 void DeckList::updateDeckHash()
 {
     QStringList cardList;
-    for (int i = 0; i < root->size(); i++) {
+    bool isValidDeckList = true;
+    for (int i = 0; i < root->size(); i++)
+    {
         InnerDecklistNode *node = dynamic_cast<InnerDecklistNode *>(root->at(i));
-        for (int j = 0; j < node->size(); j++) {
+        for (int j = 0; j < node->size(); j++)
+        {
             DecklistCardNode *card = dynamic_cast<DecklistCardNode *>(node->at(j));
             for (int k = 0; k < card->getNumber(); ++k)
-                cardList.append((node->getName() == "side" ? "SB:" : "") + card->getName().toLower());
+            {
+                if (node->getName() == "main" || node->getName() == "side") // Mainboard or Sideboard
+                {
+                    cardList.append((node->getName() == "side" ? "SB:" : "") + card->getName().toLower());
+                }
+                else if (node->getName() != "tokens") // Neither Mainboard, Sideboard, or Tokens... cheater?
```

Comment:
How about storing the valid zone names in a QSet and checking contains? Then the logic of validity is separated from the logic of what to emit for hashing